extensions/wikia/SpecialDiscussionsLog/SpecialDiscussionsLogController.class.php

```diff
@@ -0,0 +1,218 @@
+			array_push( $displayedUserLogRecords, [
+				'app' => $userLogRecord->getApp(),
+				'ip' => $userLogRecord->getIp(),
+				'language' => $userLogRecord->getLanguage(),
+				'location' => $userLogRecord->getLocation(),
+				'timestamp' => $userLogRecord->getTimestamp(),
+				'userAgent' => $userLogRecord->getUserAgent(),
+			] );
+		}
+
+		return \MustacheService::getInstance()->render(
+			self::$userLogTemplate,
+			[
+				'hasUserError' => $hasUserError,
+				'logTableCaption' => wfMessage( 'discussionslog-table-caption' )
+						->params( [$userName, $userId] )
+						->escaped(),
+				'userErrorMessage' => wfMessage( 'discussionslog-no-user-match-error' )
+						->escaped(),
+				'hasNoUserLogRecords' => $hasNoUserLogRecords,
+				'noUserLogRecordsMessage' => wfMessage( 'discussionslog-no-mobile-activity-error' )
+						->params( $userName )
+						->escaped(),
+				'userLogRecords' => $displayedUserLogRecords
+			]
+		);
+	}
+
+	private function aggregateLogSearches( $userId, $userName ) {
+		$logger = Wikia\Logger\WikiaLogger::instance();
+		$query = $this->getQuery( $userId );
+		$client = new GuzzleHttp\Client();
+		$records = [];
+		$dayOffset = 0;
+		while ( count( $records ) < self::PAGINATION_SIZE && $dayOffset < self::DAYS_RANGE ) {
+			$url = $this->constructKibanaUrl( $dayOffset++ );
+			try {
+				$response = $client->post(
+					$url, [
+						'body' => $query,
+					]
+				);
+			} catch ( \GuzzleHttp\Exception\RequestException $requestException ) {
+				$logger->warning(
+					sprintf( 'Request to elasticsearch failed: %s', $requestException->getMessage() )
+				);
+				continue;
+			}
+
+			if ( $response->getStatusCode() !== self::HTTP_STATUS_OK ) {
+				$logger->warning(
+					sprintf( 'Elasticsearch request error; status code %d', $response->getStatusCode() )
+				);
+				continue;
+			}
+
+			$resultObject = json_decode( $response->getBody() );
+			$hits = $resultObject->hits->hits;
+			foreach ( $hits as $hit ) {
+				$record = $hit->_source;
+
+				$userLogRecord = new UserLogRecord();
+				$userLogRecord->setApp(
+					$record->{'mobile_app.data.app_name'} . ' ' . $record->{'mobile_app.data.app_version'}
+				);
+				$userLogRecord->setIp( $record->{'mobile_app.client_ip'} );
+				$userLogRecord->setLanguage( $record->{'mobile_app.event.device_language'} );
+				$userLogRecord->setLocation(
+					$record->{'mobile_app.geo_ip.city'} . ', ' . $record->{'mobile_app.geo_ip.country_name'}
+				);
+				$userLogRecord->setTimestamp( date( DATE_RFC2822, $record->{'mobile_app.event.timestamp'} / 1000 ) );
+				$userLogRecord->setUserAgent(
+					$record->{'mobile_app.data.platform'} . ' ' . $record->{'mobile_app.data.platform_version'}
+				);
+				$userLogRecord->setUserId( $userId );
+				$userLogRecord->setUserName( $userName );
+				$records[] = $userLogRecord;
+			}
+		}
+
+		return $records;
+	}
+
+	private function getQuery( $userId ) {
+		$pagination_size = self::PAGINATION_SIZE;
+		return <<<JSON_BODY
+{
```

Comment:
We could, but I think this is fine for now until this needs to start handling different types of queries.